src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -116,6 +123,8 @@ func NewStatefulSet(
 	schemaRegistryClientCertSecretKey types.NamespacedName,
 	serviceAccountName string,
 	configuratorSettings ConfiguratorSettings,
+	updatedResources map[string]bool,
+	updateTriggers *[]string,
```

Comment:
New fields are not assigned to the `StatefulSetResource` struct.

Reply:
why is this pointer to array and not array? is that intentional? 🤔